### PR TITLE
Lowres training

### PR DESCRIPTION
**Pull request by dschodt-lanl**
This branch adds functionality to enable low resolution training whereby we train models on downsampled versions of data and fine-tune on full resolution data.  The added functionality includes an image resizing transform, masked losses (i.e., losses that ignore parts of downsampled images that were padded to ensure compatibility with a desired architecture), and associated architecture design helpers to support experiment design (e.g., deciding what scale factor to use to minimize the need for padding).

### CODE REVIEW

**Comment by hickmank** [comment]
@dschodt-lanl this looks like you're still working on it. It looks great so far. Let me know if you want me to review this now or if I should wait until it's out of draft stage.

File: applications/harnesses/ch_lightning_loderunner/train_LodeRunner_lightning.py [hunk 1]
```diff
@@ -103,7 +105,9 @@
             "Uvelocity",
             "Wvelocity",
         ],
-        image_size=(1120, 400),
+        image_size=args.image_size
+        if args.scaled_image_size is None
+        else args.scaled_image_size,
```
Comment (galegozi) [inline applications/harnesses/ch_lightning_loderunner/train_LodeRunner_lightning.py:110]: Why is this on three lines? Typically these statements (called ternary statements) are done on one line...
It's okay if there is a good reason to split it, just curious

Comment (dschodt-lanl) [reply applications/harnesses/ch_lightning_loderunner/train_LodeRunner_lightning.py]: That was to conform to the Ruff linting line limit.  This looks a bit confusing so I'm going to rewrite these lines (I used an autoformatter and hadn't even noticed this!).

**Comment by dschodt-lanl** [comment]
> @dschodt-lanl this looks like you're still working on it. It looks great so far. Let me know if you want me to review this now or if I should wait until it's out of draft stage.

@hickmank This is ready for your review.

**Review by hickmank — APPROVED** [review]
I approve of this. To make sure I understand what is done with the training script though...

The training script is now capable of switching between regular scheduled sampling, scheduled sampling with image scaling, and scheduled sampling with fine tuning of the U-Net backbone. These behaviors are toggled through CLI. 

This begs the question, is it better to have one script per training routine or to have a single script with many possible behaviors controlled by CLI? I'd lean toward one script per routine for simplicity but I can see the advantage of the CLI interface.

File: src/yoke/models/vit/swin/bomberman.py [hunk 2]
```diff
@@ -62,19 +61,19 @@ class LodeRunner(nn.Module):
     def __init__(
         self,
         default_vars: list[str],
-        image_size: (int, int) = (1120, 800),
-        patch_size: (int, int) = (10, 10),
+        image_size: Iterable[int, int] = (1120, 800),
```
[the remark this answers is not in the document]
Comment (dschodt-lanl) [reply src/yoke/models/vit/swin/bomberman.py:64]: @hickmank If you're referring to the addition of "Iterable" in the type hint, it was to make the type hint a valid format.

File: applications/harnesses/ch_lightning_loderunner/train_LodeRunner_lightning.py
[the remark this answers is not in the document]
Comment (dschodt-lanl) [reply applications/harnesses/ch_lightning_loderunner/train_LodeRunner_lightning.py:1]: My preference is to have a single training script with behavior defined in a configuration file (which we don't yet have).
